src/main/java/org/sonarlint/intellij/ui/panelutil/GenericPanel.kt

```diff
@@ -17,33 +17,35 @@
  * License along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
  */
-package org.sonarlint.intellij.ui.review
+package org.sonarlint.intellij.ui.panelutil
 
 import com.intellij.ui.JBColor
 import com.intellij.ui.components.JBLabel
 import com.intellij.ui.components.JBPanel
 import com.intellij.ui.components.JBRadioButton
 import com.intellij.util.ui.JBUI
-import org.sonarsource.sonarlint.core.clientapi.backend.hotspot.HotspotStatus
 import java.awt.Font
 import java.awt.GridBagConstraints
 import java.awt.GridBagLayout
+import java.awt.event.MouseAdapter
+import java.awt.event.MouseEvent
 import javax.swing.BorderFactory
+import javax.swing.ButtonGroup
 
-class StatusPanel(
-    status: HotspotStatus,
-    isSelected: Boolean,
-) : JBPanel<StatusPanel>(GridBagLayout()) {
+open class GenericPanel (
```

Comment:
I think having "generic" in a name is too... "generic" :)
The purpose of the class is to show an option that the user can select among others, so maybe `OptionPanel`?

Reply:
And in fact I think we could go further with this class, there is still some duplicated logic around it, like creating the radio group. So what about having an `OptionsPanel`, to which you could provide a model (in the AWT spirit) containing the list of options?

Reply:
Yes I was thinking the same thing for the name, I think OptionPanel is indeed a better... option :)